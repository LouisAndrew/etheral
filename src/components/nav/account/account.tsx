import React from 'react';
import { Link } from '@reach/router';

import UserIcon from '../assets/account';

type Props = {
    user: firebase.User | null;
    desktop: boolean; // two different components on nav! this attr is used to differentiate between desktop and mobile comp
};

const Account: React.FC<Props> = ({ user, desktop }) => {
    console.log(user);

    return (
        <Link
            to={'auth'}
            css={`
                text-align: left;
                display: ${desktop ? 'none' : 'inline-block'};

                @media screen and (min-width: 48em) {
                    display: ${!desktop ? 'none' : 'flex'} !important;
                    align-items: center;
                    margin-right: 16px;
                    margin-bottom: 0;
                }
            `}
        >
            <UserIcon
                className={
                    desktop ? 'icons black-on-dropdown-stroke' : 'icons bigger'
                }
            />
        </Link>
    );
};

export { Account };
