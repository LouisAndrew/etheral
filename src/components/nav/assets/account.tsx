import * as React from 'react';

// eslint-disable-next-line @typescript-eslint/tslint/config
function SvgAccount(props: any) {
    return (
        <svg
            width="1em"
            height="1em"
            viewBox="0 0 64 64"
            fill="none"
            {...props}
        >
            <path fill="url(#account_svg__pattern0)" d="M0 0h64v64H0z" />
            <defs>
                <pattern
                    id="account_svg__pattern0"
                    patternContentUnits="objectBoundingBox"
                    width={1}
                    height={1}
                >
                    <use
                        xlinkHref="#account_svg__image0"
                        transform="scale(.00279)"
                    />
                </pattern>
                <image
                    id="account_svg__image0"
                    width={359}
                    height={360}
                    xlinkHref="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAWcAAAFoCAMAAAC8MMpqAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAA2FBMVEX////9/f38/Pz////u7/DFx8m6vL6UlpiTlZhiY2ZVVVdVVliHiYutr7HGyMr9/f2foaR8foBjZGdhY2WeoaPh4uOsrrB7fX9UVFZ6fH+5u73g4eJvcHPU1dfv8PDi4+Tu7u/V1tfT1Na4uryGiIuIiox9foGUlpnHycugoqVwcXTIysyrrbDV19i7vb/j5OWusLKho6VWV1mSlJdISErw8fFucHJxcnVkZWf8/PyVl5rt7e6Iio0jHyAkISJGRkg3NTc1NDU0MzQmIiNHR0k4NjhFRUcAAABhf0mRAAAAPXRSTlMAAAAgMGBun5/f7++vfmAgj7/f4ZFAfr/xwXBAz1AuPjBOUHCxr72dXo/PXoFObj5+je2h/S7Rzd0inTKtf67eoAAAAAFiS0dER2C9yXsAAAAJcEhZcwAACxIAAAsSAdLdfvwAAAAHdElNRQfkCQgXCQxIuPs9AAAc40lEQVR42u1dCWPaOLBGO26AEHAgC7QpmJDdNIRs2t1e2/feWkBL+/9/0rMOXxwe+cC2jL825MCH9DEejUajmUajVCCHfiP8nwMQMAz5Q/TZNY6CBF7lT4TRa7y6aLba7UvbpjaDeO2021fNbs9wCSeBq9SUR4KEWCYNMHvX/YFNqc+u/93DTf/3HpjeSTuXqXEAhLgMAQxH49ecVhrglv/mfbkvzrfOm+4Q4Dd+Ee+lxgEQ/4k3jdvxSnLqckw9esN/8IWcrq4uTNhVPjUOg8Db60lQeAOkBlUHDfBt+6plMjIgfSMqDzBGr4MyTEP6IQRqh4in3ndOdS3QR8Ati9vJHpUhsQ7RvTMwBg5aj6e1UIcQEDswWis7PSTdlyOAgKl35uLt27owfEPTERwi2nnZtCwgwRudJ0jAACPwqp0NySGZ/t6fwTlT7EHaYPDqxiMovVCHxsfxXc20qzHmN/sEZSHO/Pu6ZZw700xzEBiODxlt2dDNrerNyGycN9PEseRam+wJ3lU/nXs4c5p7q4ALjmYp1eGH5Mo6X3uagDFJzmNMrHogP9uz0SBuP+GPU6iMo/jTOj81zebYOQqzUEirubn7UVcaYg1qujqFiRHJNL0G9/5nQjSB5kksuaMky5eJUXTfc6W5YeWnM4LOVWpfnpEfj8D95Y6w5SXPzqSlB+Q8rA4Cf6xzlOa9j3N0JhINo3yk+CiufKKrKtRsBHzIxCmXBm+ABGM9KgkY7z7HBWAMjYozDW9clgsk2jE7Km3fkcbd2M7QzZwCDtGEy3Qlxdno7ERdFCXPnOhqcux0y+qUQpYFuOqo4jo4cYfAkhA9gUYlRdqhOR3B9MBfQs9HTHtxXLkJC/FpVmHCp+9HkOPLwbvWY+tiseg9NZut5WDrvpH043tTOaIdmvuhGHEVyfWPXC9bzzMeXhQMXwLodR/a1PUTxWeb/lUtogmfbKvTHGJ6PXl6MSEU4y+/hB/b7DV9KyYe3ZSOKrY+C3+FCcRpFvpgPX4P4JK645/3Q7rgQ3cZ7/ruoZtquUnhfh2LBimey7+De33I/k++HoHh0za25nBus6nQxJA0rLX0tsewCNYPw3Bo/vHlJrkZazGJraWdGXhlBJo0jEsRb6/OAd3+Y3lrecqAYT+uQIe8pJoDxkoE+3t76LZrJuo9gVl/54o4bs0ktyofmF9fkWgh8+umlWzbA/MMwWwZU3d8HzaqMC+EaYyHmH17JzaXxO65ew70PoY+ORSvQXc/B2u8ob4XgonzagGBTYTJbmo+reM4Bekn7QN3nT7HiiCgLUi3U0qca84GcXQHvdeZZvEQXyiLlYPPUyApd7NK6898Ug9A1d64c/o73Kg/v/Y7K7gnNs1tnedo2onhv2vpTDQTq3YMmrvgnZhUQQfPspYxHqVp0WSlIhq6Kn0UVKznXoByRjCfwnc4enu2C1/r6Gjru/Jw1PkCGacZcHTHhbpb5Rr0pVlxIsgwsLKPliUNmK8VabbXd3pyzNBTJJna78B3K2cJmK9UiGZvP2g7/YaOqjT3/REwS7D1BeujJBKleph8+C2W5pHiI+vTnDlYqHUHY1m+19Z0bYVPuNVodv3M2XdUEI0wLd/qFc1YIkBLRTMGpPkE0iQuOV9HN8KdzrS1nKwYG5xk9n7n5L0DjGgXC81MDh6u0VKhmdod6/TtMZ/9oL4fEe1p62VycGvfEWdqR/aK07x+yaNF0FVyJzkCrZU8sxlCS0Ezs56dXieyjx36+8E3B5oz0E5Dm8zY+IEy3cylY4SHqmJ6jD1ew6J5i9Ur52uEP6cOBjmlP2MO2jXqJnXeH2uWjw2Ush1tchgDGYh0HSooac3CZqZKhtRFPtIjswMtUc1hS7edNvAiyiMxybdPX1R2h66ydoKfFBCZWMPdAfwl3y7B3yrTb43mKoTcqsxR/okf2JUOMED9dtQea+RNggnqHyvEVp3hGppuZJCODmzfKShCumBH5tsbeFBwuIwa2mRC6UaaqiJWcQkFLMYZ62j1zP5/BU1Y5tYGakDNcu+LzGKDQhsT2sTTeNEi9pQxn3+0bcel47Zo/lQxVRCaghwJXKCRZ02b3W5NfM3z32L8CAQTaIaNJtGj3LeBeGwKW4qDvq8gjmGqBc0NwAfBTlHi3CAzfAvtddEMqmGKraPYdrcQgRGVWQao4rjRQ0E/4fJsFTThYvfsojzbejihr9B9If0C1zuJtVPy5gDudZh4K7j4LwrrBPFGwihcl5tj6U9Hu/Ezb0ddGGwnR7Sba1w0kwi4Y2CI8rwsdpyxJJvH0dHBN/oK5blbbANhibaw7PEyzEDtor34UHArn9Cpd8kjGvnT1sJo3hYtLS9IAyn3QZcXQqldYTz3i/YfAOZPpK2iqYwGNzpRs65g9exMCb9hTWwX3UQETExRnnOJXIwEqtpuyj8jhEusE8V7Dy5QBV1mw05MVSG6A3nElaOY21h4QenlmUA0y9QeFN1ER0EjNP+wp+WmmVU2j5QTp3vNohvpPHM/EbVRcgOaaQ7UvVEGnpGxmtJXRTcxGoQ/k9Gy8lx0Ix0MIv2iVIclFVSeF0W30MEDNvEu+USlgcpzOXhGw2XKz/NxeRalvMrCM9WZZ4LZdbQM0xT95ZlEO2m4DBXOsyMMaCRPuXlmxr3Zjqa5eJ6dZuquNzB/HdPPsxK0sYkFmJSaZxLtr5NhQGUZB6muPMuc48edjiKqoyQ86y7PBHPuloFnNISj5DwzROw45o/qRdHNdNo50HwcZPLc8xg9rDda7pHFgccyRhJd+ph+0jDsCKIdPDSKDuMmiA+mJIN1RAd45o1Q2dsdlqnMUUCKXfDG1DPtlX09BQ8w3p4u7ZQqehQJsLNzSqSQAqQxPt58rqOLL5PWtbFIxjKvw7po0Qiebb4mVHAvHjC18av0PBOeVfSosDCX3VOj6HgkdOsES/xVdqa5wRGlAE+X5VIReIT2VdEcKnXjdbT625rFygp5DqmxQyj5NEX2A0smMyv4kcRSvlE98tihub6eCizo/JvzvH1E037pUQsZ2zmxLHQ0JxamnVmuEx0Av5BVocJmAb8xOe26YhshCDqIM985FuVGp8Vsh+UgLKkQtr9Ri2GwwV2jkXlbloXtnCByu1V0tGjJo748GJgGLGiNkD9G/2CNK8NSsRpMLNb8qaCRkKgUZ6A3xe4jjdEbbP2twC1XCrVGmhpwzEGm0QJD7ffFtMsZBd9Fc8ziDe41odl5OJHajrSwGhozVJp1sZ657fQnloKjoIUhx+SMatkPkflS9qLkIIjioEKgC2kZmoaK8gX58mffkIBVCQWa4Lk3qEj3pQvNDbjGeP6Wm0AHOJth2XIdFJHBMHnPhqjv8T3kXu/P/BcX5/KHbnjg5crbARE5IDW2vbLyo1jq2wVeT4xu9RFn3im0WBtt5uYVk7cxOwpVAz5pwzPx4voRt1iOtW95BPyjjfi32JsaVVDhwUZwdbxX0mma205vWbZNpc70pPQRBbsdOz7z8jqbTzUgCbNzbMAIotwbjg8BkI0q7OUlP6LNpgrNhecUSoAeXnNim9cKFoEFpjE4RnqsWIUA22gBYu/kksqOWXWGSpEat0qNXgAsvoCGjLuTyRGj2VIovcxqTWg1CrrA8/Q7XevmIUIKOUVdcdaQaGjhZbPtn1M29JzWQQbo/ld9tTPD3YZiXg6H6JdTSxF0FeqkO/ilo3ZusEEe2eMmus6rEJ6QaFEcVkGgezp5REMw8EoqDgOXJ7XuWIFH/0ONgJ7luxmIUjVe2/5ouV3MRJqI/wNh1fAU8bZoulL0+G6tVPZ4/eKtyWXBNPEvZapUlOaN+AT6LFftdRfczSpIecX1HAgR+93Sd9WrA0YIPKlK8y9LT5Il7ja4buRWxyjLQch9LqCvIs2UWUW905uXJ4QQKNQfyd7OznnncgUGXivFbcCVnlMU2V8iC6bjNDv/l1+ycpbJggzzrZLZzI5ZGe6JOpLN/f23qjJlb6fZzMckzU/+s4Kil6nBUwjUSqYLZKc7wBrEoLml6YQ7BLmbWqnDnXmYaZXekwMHwoWSH1S2Sd8ZSqjPbxSlmff6EWKb0iTINddVMzQzvHtDdk8+8ddfoIm1URJnecT2IoWFx1m2HlVmR95DRueVEGfH5EDz4YeZHjz/jyekaoojYPea0N0qKil52LXGJl0YeM5+t9+Sn3/nsEMfzjV/AYdl1eFAHjHWZJeECgWv1KxY/yg6eI7xMMvPgzgsr/YuheCSWc7FprfJjmjlodATM2p3ulaMB5r5Uj5crYOCqnSzzbSh7fTkAAtqjuid39f9BahKNVjdQRwxdnFdkTFQwsTWvo9g+/AcQbVryJlWF6/BdhCTStHMHup2AmETUj15mgPs2h4y44GjkqxFs2PHUhY+NkbR1GRNNPngao4EhFD721V3YUHYzwTmh97jOy9qLvZ12VaUKsy3wzCfEnDsjonyvI+Db63HVut/W4/N/mDw2T8s2ZMiNiNVjmhlZ/Ae0cgBNBnVlMU6V4ll2Zehis1xkGUa+Np9V9lFtY/KzFA8msWI9buShB6W54O5NOmRnxWhUei+KtH8NY4nOiyybmXzMO1H+T9LcQ5g9j0RIacAzXV7TM4w1dewTo+VJtu4kwDGCYesbOHte60kyaxXd9yf9iMdTVkwTVmyKVIVT90e0TBNbIVljZb86CtHNPcSwyduNRQt0bKuZ+U4lkQzX3y7JBI9BlJJcXZhbIomWoa3S+dGlYgOLPVB7zLY2wIxqpTvuREQGmccfDtINYfLEgHnc0XE2jWe4K1MDV08yQyT6nmfOcvGp4Asl4HqcfElRjKEzMdhttZhT2cJwGwOT6fpD0bz7bZoUvfgfNiXllkZ7ewA7ts02L9yiDNrxfdFddwcYIxTeuRPRjOl9pWh73DoByGShgnXavGiRWF1C+GmawS/teZi5QpPafF1CG7IjVY0+42F4aQs2jgC1G4Zupp4vMHm9UbYyqWnevMX6Li+InygU4UsLmUAl4Jft5pFzbjZD42rQC/KDLeB7Z5eniWx5be3OtSXckIqNvp1qFleNTAmQYZp2TW01zpvQCw3pK8A/ogd4xXJguLuqSywGXmWR6wNMrlzzIT5a0bdpp6pctDFtx+elMEd7fY0uAm5lEQLM+P2e2b99v17hwuCHwgJS3MveZGrO+n0KGXUgXjIwOhnKmDHmLeDMp7N8rl/6e0fpuhRo5QCzWh+tc2Q5kh3tWskZHU36gVMOljeldeaJoHUgJky7VGZZHtEPKK9629uy2t4wNDPsZwBH0FWB4NB00H3eeHh78cm2z6hkBgoGSZGOY1pAqPY+yRRDJbN7mIO5mHhYk8QwGzRbS6T7MpAsLktuDLzYRjj2Czvb4TwNfK3d9cLS3mfpsP37Lm13AZtwfQf+liKdHnYJjBV3Sx/nAL3Dx/73RdIsp4EYLxvDoKfWcpnazWFchkdMPoet1+evRBSxJcPF198oyrmhy2aYi6ag2BLaGKbxDnr9xLZHUQGkMcXoDAB26sL4V5INj/wTyIAFw9b7x4JzWsx3Z8YUJLpN4G3l3YimgNyQ5dPw6CuSMk00yHTfzr+9ZM0SnRoNS+J3WH2UoeALnkSCOlQSL5MR8JzOAKzp04G9s9fpTClAzuKk+0rXna/QFCOE/vJxLmh+CJGtWq6wKMtpG9KQDSMd2dtsToh0pk0QiKcrDLCsfgtgHlfKcvawRaKz+iyCLd0wF1IwIhhzgXliv+wfpjl0n6wRt9CzYiN1X3uRAcfapiulJtKw79Qu92FvEYY0jBf+oE4ylgmCD9p80f+RHvPNfRirJuERflnf5ZXsRL5/IHV+pxmojjK3+yQg3rMbDx+HD/dNq0cwyW8m0D3s9uGmO4BdnhRFU1F9ehYM0Dx8lkojLzM/6BJbi4SOpxcovMsAiq+JdpE7Myxtl1IZlCkbrYIeFgkSQwmJOoy/3xKxJrEbKroXXsBfq7FfOXZc4D0BkmMDnbGaUuQHKQ5tnuO4XMu5ayiyWZML5JEpHGiTy/RJPhDHLPZa2RxLO/1hGd7jas/2HgoZiynfRD9CI3ppXqiVPkzXTdL4o9piIyvrXXMXQbiU9nMT22PegrVnCtGaFDP6WXTB7nlpnh49vQ4vjwzoqcnd5S6hRwU8+8EVqkH0/zLz6O94QNifOfM5tQJrOROwHv1NEfSjbd+b/qXKB7E+4In18OET8RpwKt06n5w3aw813b9Rg/WXn7Q4okWAGsZePIUJYe+Pr3VYc5jhoLSTo7Vo+NRLdLqLbbxJy2dUxMNcWjmn33Tt4PKI9CNgHcXrKv4k5aTES05UlcagueODsUczGf10gluz8an84MRPj1Rtej412N5TOaofoHVj5G0iZ6MaPdi/6c4C5Sm0ueTm/SZwbz9GWg63jnn9Trz/cquKjPUFo/d3SjvdMkWzsZDXhArTkEQKpbBTyBIMFHUYmLtrwt6FU6EZpzqNjabGJqZ1P/cbUc/hm7WYwD0wbciLH7GJDrTB1bmb1asXSzR10VnhIhWK8BJPc14eZe5OEOMtUBqd4NbljQCPCh3kWOc6VjIZ9tKdxehf2u9dIbfS+aXjhdScw2Z7cniF1KZn7hqpWNonAtHVJVVx32GCSVI4+5S3dDoW7qyLBKgfumoJjLkbv/sBiLCLTpVPEAgmlAruNMERcNKmtHtrNL8E/VCjc69RxrrDJdo8yGO2z+zIr48rb7afTfvvRFQV7L5qK9iXHmhyBmFOJL5RnFdh67nps4MewFtbIueuurY3mVyb+O16hMUsOd05dorzTmPYd+N03ol+agwVvpQHXRyiwA9KdzF5hjLcyn3KHOaR0p7IphL3wo0U2dIiX7ZqHrNqGvcJVvQ56eoLaBwmnVyzil0nRGtKM/2Mq3m8EpCR+TOEf45bWcne/DMu5lyPWw7VVI2Z4Jypag1OnkHUp4UbtywoVTYnmHjhnQk4hrcwtuRZp2rm8sVo5EF1KwOTo5IQp+s+8RauWs50UT/FLXPqkOym+zLVCBaLnb1kk8L4Yp6SZ6inh89/aAI3JBNNTua2isjgaCJLKwLNd1USZp9O1pVR7eSsMCXt3+hzwtHF7R0zylx4NjRF2rSZttvvZMULy8fmQe1D7IkQfonAbeju4o8t2O6KuXn2LMVJoE27VeWZs8f/ahI9G0sot3iPUr53umybBmEsoR036EuHolVPJETC5IqYQRUTgOrybJPCFwq+IWdAz7Ff7aHG/TSztvrYYp0GRqAyDSjlsriLE2wWGiqrAhSe1FZ5exBPN1TNTP6KiYffBDEhNm2m9WnmcOZGHaD/T6OoXeGEmCLMMxjn/7Vt5JOTBA3UglTH+NYe9fhKfKC8r2tbjF0icGtu0vczKVMoNUTCxJjg7Ns289Q4REwSIf4Zq1V5hNe8UicGMK9zpF+ffbm45koZ8GJI9AXSmHoPXm4wiWH2LXYzQZJs1NqCmcs/A+ryMDevVEWP3OMs0zXH86GYyJiOR2iP2ICyLiZNhTFb2grxMU8nZPWcDFHebZlGiUFor3Ef1EYnCHNRKSeRDFUsw5U/HTrWdGdLgYw8DRnlECrrJSKXVUI0WepNRhmaxtn+q2K3hjKy0RdaaBx8G0aOJrjUUGg8cQ5nq818kJ0XnSHC4G0OXDP3XeDINdp4Lazg/902LV9KkyFGEay/Rfm+CHwBjc21syvcX5qw11ceUCIpioLK0bg6GPQbUNxxrAUXNG3URcgfNMris652hoS0MWnFxNEswIWiuo8L4szlmVBEpq2gdqRKcFIA9//QgdQ5QVBJSyi9Sp7L3oBC7CAX0ecP7ADz5rmhvkNncmto3gmPYRj2985ftZMD3H36PGRUC0eZCgOPW/AOyQenLJ1laMwcNu5ulFesTBD5TliJIRrTDnb9qwWZgZHoDFc70tkKKIuUj/X4iwxQ3le7fnaXCtt6sn8UXme19LcEIuy/SifJjcZpuTIqWMbc5AMKrHhNQuQFyQLGz1kQosYMmQu6HwEF7U4u4BBJFNCcezqDf51i7r3t+fsD/Uh+LoI1N3Z06/uwjfZOa3RECUxo937T2e6jLIHnmfuaAii5JG+OahljQ02mVwn2blVUbBU6L7w7moNmcgfdk7hGNnYLKU26jwwzqxjVoP3t+mBM6NTTfFTX4ruXbnATbtIyfwTdhQ0QawNfrVtrZ1DIM82gu2OxcF+uY0kmX09pdgpXkk4IyHihd6fqjgPQbQ0c9dGjSCgiQ1prZ2yt87o+T2aZmp/rNXGLoY2QvTrXYuD3GO6xn4SR9bwQACr+kaDzlGudJvICTat1cY+EEeybY+CK6nM2kDztHZkGs1aon2QD5HuTZafKqA4eAoINbVRIwASvf2Pa+7wlJA5RRCtUVsbhwBXmHiGIz6xkrqUT1JqhEFEJEekQLdCJ0TLP8NDzfM+HHMYW7huhxR0ZDAud/NdFN2ncgKWCHN2SD6jw70Y0WezxTgOHMURtUGbv/MqeAKqzs89RvQgmJEbFcHFvXnNwAnOvCZ6IcV+KLpPpQSz7DAJDSpoAw1Duk3emGoD0ApCgdTjaFKT2no+CDbD+y9yXHNeen59eTSIf12r54MgpDHCVMF1w8tU9xXj+Rx3GSsC2fJN7a/eqreJ5u1+rP1Hx4AOhBvTdSdbWNwGk/2a513IPK+YD9qeudS9wo60FzXNB8FYOW5wyHzOnq3WVOC5JvoISB8jz1skxFICUvtL0b0pI6TkoVLqzlQAr8lt1uJ8GESBZ9fXD1j8opjT1EwfBq51ZUq7KZp/n9YhBYdB8BVs23NaXOB5cRc1y0eB74hvCfZaKM30uejOlBfkP5TntgizeyPIjPxEiu5NOcHTjrZRnlcQODCS6E69/2cfcj6okOecGxzmRiFd3Wzn6v5v5GgLtAPB3j5UvUSl6gdfuwKFYgd0Ajvlq1yC42Q7LjmOBB2TQ78S92jcz++gyw6dMpp/RAs+y9UTagYJE70j1r+l6WvG1GV4tCdoxPuuUsSGCoOjp1SMZr0IRJkHJPuA4kgh4hlnUElWbur478TvsFuU6acKe7yAzTUizS6aAIF7ROWSTpifI+vNAola8NvuH0jEwXCBpt7m4Om/FGugU3s7AjPipum7me68E1zqmBi5wqyimzkYzyq1buS+onW/OwcINkHe04QdvCwSANhl9i6VBvPF4jlOC3rO1zx4Ad6aXZIdtQywaHZUKo0KsA1uuJ0dRmewfGwKDAaD9oDj4HbFJEh2VjYXO7AvnpXiGXhwusy7PhmolK0J4JXSfKZGWjjzafiV/jI1IkEZz2bRrag+qD1RWBevkRLU/nFT85wL2kTB31QjNQZQy3Me+AUNs5bnHAC1vZELar2RAyhz2NU8nx60jSa7rJEB6E3t38gFY0W/aI1UoCwuA80hUSMlxG5ifLNVjVSgIuc2uL/UyBySVb4+yBME1jSfAm4aQZGhGK+ZUiMdRP5LnlKtFujToS3CBMwWWnS3RhK467o9GZd3V08JTwF3k0Tb3agJt2kvWeMQZBbooRftYj7g9bprxAen9CIQxMWzuNKdA6JOLhnK2CaJbiicyfSJplHhTIpin2+/c2ZZiQP3kBDNjvqQldYp3T0wwU0y63feYqp4v+M1U+jOUdv73YzmDZjd0NB1aLx7hPAjI44iK8FkDsW4RHpcBsPYfLrbiYEUSrr3lfpdoypXwtqcQcdLR3PkUYG/bz5ZcCBQnZcEeXvdDgu/6vXTHVqGm8W8SGSdCZuuxrcyLP9gSSCHali0Ju04V9/HZfvm5mZw006Jwddmq9X61MLxqZnB7doD9nKjsLwUNUZRp/et0dALmj60Pc3bD2ECGL19vOr1Fkrh3SnD6AuFagh77yCGovf7u2z+Hwe7Piyc90W6AAAAAElFTkSuQmCC"
                />
            </defs>
        </svg>
    );
}

export default SvgAccount;
