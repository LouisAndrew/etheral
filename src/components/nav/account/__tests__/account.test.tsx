import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Account } from '../account';

describe('Account element to log in ', () => {
    const elementNotAuthenticated = <Account user={null} desktop={true} />;
    const elementAuthenticated = (
        <Account
            user={{ displayName: 'Jane' } as firebase.User}
            desktop={true}
        />
    );

    afterEach(cleanup);

    it('should always passes', () => {});

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(elementAuthenticated, div);
    });

    it('should render login button if user is not authenticated', () => {
        const { queryByText } = render(elementNotAuthenticated);
        const LoginButton = queryByText('LOGIN');

        if (LoginButton) {
            expect(LoginButton).toBeInTheDocument();
        } else {
            fail();
        }
    });

    it('should render display name if user is authenticated', () => {
        const { queryByText } = render(elementNotAuthenticated);
        const DisplayNameEl = queryByText('Jane');

        setTimeout(() => {
            if (DisplayNameEl) {
                expect(DisplayNameEl).toBeInTheDocument();
            } else {
                fail();
            }
        }, 200);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(elementAuthenticated).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
