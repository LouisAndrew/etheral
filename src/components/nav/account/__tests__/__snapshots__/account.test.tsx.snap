// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account element to log in  matches snapshot 1`] = `
<div
  className="css-4vz6jx"
>
  <svg
    className="icons"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<path d=\\"M20 22h-2v-2a3 3 0 0 0-3-3H9a3 3 0 0 0-3 3v2H4v-2a5 5 0 0 1 5-5h6a5 5 0 0 1 5 5v2zm-8-9a6 6 0 1 1 0-12a6 6 0 0 1 0 12zm0-2a4 4 0 1 0 0-8a4 4 0 0 0 0 8z\\" fill=\\"currentColor\\"/>",
      }
    }
    focusable={false}
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "transform": "rotate(360deg)",
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  />
  <a
    href="/auth"
    onClick={[Function]}
  >
    <div
      className="css-193p5vv"
    >
      Jane
    </div>
  </a>
</div>
`;
